workspaces/ui-v2/src/pages/docs/components/EditContributionsButton.tsx

```diff
@@ -3,20 +3,60 @@ import { ToggleButton } from '@material-ui/lab';
 import { Typography, makeStyles } from '@material-ui/core';
 import SaveAltIcon from '@material-ui/icons/SaveAlt';
 import EditIcon from '@material-ui/icons/Edit';
-import { useContributionEditing } from '<src>/pages/docs/contexts/Contributions';
+
 import { CommitMessageModal } from '<src>/components';
+import { useSpectacleContext } from '<src>/contexts/spectacle-provider';
+import { useClientAgent } from '<src>/hooks/useClientAgent';
+import { useSessionId } from '<src>/hooks/useSessionId';
+import {
+  useAppSelector,
+  useAppDispatch,
+  documentationEditActions,
+  selectors,
+} from '<src>/store';
 
 export function EditContributionsButton() {
   const classes = useStyles();
+  const spectacle = useSpectacleContext();
+  const clientId = useClientAgent();
+  const clientSessionId = useSessionId();
```

Comment:
So these things could sit in a `metadataSlice` - which means we could expose them in selectors, but we wouldn't need to manually pass these in for the `save` action (which instead we could derive from the current state)